django/db/models/query.py

```diff
@@ -473,6 +473,47 @@ def bulk_create(self, objs, batch_size=None, ignore_conflicts=False):
 
         return objs
 
+    def bulk_save(self, objs, fields, batch_size=None):
+        """
+        Update the given fields in each of the given objects in the database.
+        (Doesn't call save() on each of the instances.)
+        """
+        if batch_size is not None and batch_size < 0:
+            raise ValueError('Batch size must be a positive integer.')
+        if not fields:
+            raise ValueError('Specific field names must be given to bulk_save().')
+        objs = tuple(objs)
+        if not all(obj.pk for obj in objs):
+            raise ValueError('All objects must have a primary key set.')
+        if not objs:
+            return
+        fields = [self.model._meta.get_field(name) for name in fields]
+        if any(not f.concrete or (f.is_relation and f.many_to_many) for f in fields):
+            raise ValueError('bulk_save() can only be used with concrete fields.')
+        ops = connections[self.db].ops
+        # PK is used twice in the resulting update query, once in the filter
+        # and once in the WHEN. Each field will also have one CAST.
+        max_batch_size = ops.bulk_batch_size(['pk', 'pk'] + fields, objs)
+        batch_size = min(batch_size, max_batch_size) if batch_size else max_batch_size
+        requires_casting = connections[self.db].features.requires_casted_case_in_updates
+        batches = (objs[i:i + batch_size] for i in range(0, len(objs), batch_size))
+        with transaction.atomic(using=self.db, savepoint=False):
+            for batch_objs in batches:
+                update_kwargs = {}
+                for field in fields:
+                    when_statements = []
+                    for obj in batch_objs:
+                        attr = getattr(obj, field.attname)
+                        if not isinstance(attr, Expression):
+                            attr = Value(attr, output_field=field)
+                        when_statements.append(When(pk=obj.pk, then=attr))
+                    case_statement = Case(*when_statements, output_field=field)
+                    if requires_casting:
+                        case_statement = Cast(case_statement, output_field=field)
+                    update_kwargs[field.attname] = case_statement
```

Comment:
Pre-computing the updates by moving lines 501-513 before line 500 would reduce the length of a time the transaction is open for, which is a good thing for database throughput